Firestore/Source/Model/FSTMutation.mm

```diff
@@ -505,7 +487,7 @@ - (nullable FSTMaybeDocument *)applyTo:(nullable FSTMaybeDocument *)maybeDoc
                                                           writeTime:(FIRTimestamp *)localWriteTime {
   NSMutableArray<FSTFieldValue *> *transformResults = [NSMutableArray array];
   for (FSTFieldTransform *fieldTransform in self.fieldTransforms) {
-    if ([fieldTransform.transform isKindOfClass:[FSTServerTimestampTransform class]]) {
+    if (fieldTransform.transform->type() == TransformOperation::Type::ServerTimestamp) {
```

Comment:
I think it's likely that other transforms will require a cast into the specified transform type. While this is not a huge change, I'd slightly prefer if this was still an instanceof check.

Reply:
I don't know a good run time check like `instanceof`. The virtual method `type()` supposes to provide the RTTI as any subclass of `TransformOperation` is required to override it to return a type enum. Required because `type()` is deleted from the base class.

Reply:
typeid can get you there. But I don't believe we can assume that RTTI is enabled (hence emulating it via type()).